src/pieces/king.js

```diff
@@ -24,7 +24,10 @@ class King extends Piece {
       const newColumn = this.column + directions[1];
 
       if (newRow >=0 && newRow <=7 && newColumn >= 0 && newColumn <= 7) {
-        possibleMoves.push([newRow, newColumn]);
+        const targetSquare = board.getSquare(newRow, newColumn);
+        if (!targetSquare.piece || targetSquare.piece.side !== this.side) {
```

Comment:
tu można fajnie wykorzystać optional chainging wyrażene `targetSquare.piece?.side`  będzie miało wartość undefined dla pustego pola co też będzie różne od `this.side`

Reply:
done